core-bundle/src/Resources/contao/library/Contao/File.php

```diff
@@ -349,10 +344,10 @@ public function __get($strKey)
 				return $this->arrImageViewSize;
 
 			case 'viewWidth':
-				return $this->imageViewSize !== false ? $this->imageViewSize[0] : null;
+				return !empty($this->imageViewSize) ? $this->imageViewSize[0] : null;
```

Comment:
Why?

Reply:
`$this->imageViewSize` can be one of the following:

* false
* an empty array
* an array with two elements (with and height)

Reply:
I don’t think this is true. `$this->imageViewSize` can never be an empty array. This is only true for `$this->arrImageViewSize` I think.

Reply:
`$this->imageViewSize` will return an empty array under the following conditions in this PR:

* `$this->imageSize` is not falsy
* `$this->isSvgImage` is not falsy

Both of these will be true, if the file is not an image.

Reply:
Without this change the `ImageFactoryTest::testExecutesTheExecuteResizeHook` test fails for example. But this is actually another problem: within `ImageFactoryTest::testExecutesTheExecuteResizeHook` test, `$this->imageSize` will return an empty array, even though it is an image - because the service container is missing the `@filesystem` service, which is required by the  `@contao.image.image_factory` service, which is used by `\Contao\File`. This throws an exception - however, that exception is simply suppressed within `__get('imageSize')` and thus the test does not actually fail.

The test did not fail before this PR, because the `getimagesize` fallback would then be in use.

Reply:
The question is now: since we removed the `getimagesize` fallback, should we also remove the try-catch block? We do effectively rely on the image factory to provide the correct image size. Any errors occurring there should not be hidden, I think?

Reply:
> `$this->imageViewSize` will return an empty array under the following conditions

You’re right, I missed that 🙈

> since we removed the `getimagesize` fallback, should we also remove the try-catch block?

I don’t think we can. For BC reasons accessing `->imageSize` should not throw an exception I think.

Reply:
What if we at least moved `System::getContainer()->get('contao.image.image_factory')` out of the try catch block? That at least reveals the issue in the tests, that is otherwise just hidden.